packages/block-library/src/query-pagination-previous/edit.js

```diff
@@ -2,21 +2,84 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { useBlockProps, PlainText } from '@wordpress/block-editor';
```

Comment:
I think these blocks already shipped but both previous and next bocks look very similar, could have been variations no? or at last they could share the same code base (like we used to do in the past to generate embed blocks before refactoring to variations)

Reply:
This had been considered during the implementation, but due to existing conditional logic regarding `inherit from Query`, I thought it was best to leave as two separate blocks to make them more manageable.

Regarding sharing some code, that could be an option yes.